### ISSUE TITLE
Plans of further vCenter integration

### ISSUE THREAD

**Issue by bb-Ricardo** [bl4ko/netbox-ssot#339]
Hi,

I found your project as someone mentioned this project over at https://github.com/bb-Ricardo/netbox-sync

Great effort to integrate so many sources, a lot of work. I was wondering if you plan to sort of match the features netbox-sync provides? If yes, maybe I could support you and eventually retire netbox-sync. Seems silly to have two projects working on the same solution. Could provide the check_redfish resource as well.

What do you think?

**Comment by bl4ko** [comment]
Hello, 
thanks for reaching out. [Netbox-sync](https://github.com/bb-Ricardo/netbox-sync) was kinda an inspiration for this project. The main difference is that `netbox-ssot` was designed to run multiple sources in parallel to speed up the sync process. Thats why the **go** langugage (not python) was chosen for this project. However, this also introduces a trade-off—being outside the Python ecosystem of the NetBox stack may make it less approachable for Python-focused contributors.

Regarding the features of netbox-ssot, most of the currently implemented ones are very similar to the ones in netbox-sync (maybe only different naming convention), the difference is that netbox-ssot doesn't currently have as many features and so it is not as configurable as netbox-sync. This is because the project isn't as mature yet, and the extra features that are not implemented yet, didn't seem so critical to me. But they can for sure be added in future releases.

Regarding the check_redfish source. I implement additional sources, if I have access to them, or there can be some kind of mock API, so I can test the code. So by myself I can't add it, but if you are willing to contribute on this topic, I am very happy to help implementing it or adding you as a maintainer for this source on the project.

**Comment by bb-Ricardo** [comment]
Hey,

Yes, reading through the source and the config vars it seems that netbox-sync had an influence on this project.

I'm aware of the serial approach in netbox-sync and being single threaded. Getting this right with the async framework seemed more trouble than worth it. Also some resources depend on each other and therefore it seemed a better to collect all the data first and then syncing only the differences.

[Check redfish](https://github.com/bb-Ricardo/check_redfish) just exports json files and they are ingested one by one, read from a dedicated path: https://github.com/bb-Ricardo/netbox-sync/blob/main/module/sources/check_redfish/import_inventory.py

netbox-sync itself did not have many contributors and therefore I don't think it would be a big issue changing the programming language.

**Comment by bl4ko** [comment]
Agree on the aysnc part, but the problem is that larger deployments with 10k+ objects, just work to slow with synchronous approach...

Thanks for the resources for the redfish. Since, I don't have redfish service deployed locally nor I can't test it, will you be willing to maintain this source? I can try creating starting template (code) for this source, from your python [example](https://github.com/bb-Ricardo/netbox-sync/blob/main/module/sources/check_redfish/import_inventory.py), but I would need someone to test and maintain this?

**Comment by bb-Ricardo** [comment]
Hi,

Yes, that would be great. I also started already on the structs and some parsing.

Still have to understand how the internal inventory is used.

On the other note: do you plan to add reading of tags from vCenter?

**Comment by bl4ko** [comment]
Hi,

if you are have any questions regarding the implementation of inventory, I can help explain it you.

I will add readings of tags from vCenter, I will open separate issue for that...

**Comment by bb-Ricardo** [comment]
Would like me to open some issues regarding some topics of vCenter syncing. I don't want to just create a bag of work for you which then seems like an obligation.

**Comment by bl4ko** [comment]
I’m always open to implementing new features, especially if I can test them on my setup. Regarding vCenter, I currently have multiple deployments I’m testing with, so feel free to open as many issues as needed...

**Comment by bb-Ricardo** [comment]
Hi,

Also here just wanted to let you know I'm making progress, very slowly. First I had to statically type the redfish inventory json structure, which wasn't the case before. Got released yesterday and now I can create the datamodel and add all the parsing.

Will open a separate issue for that.

**Comment by bl4ko** [comment]
Hello, thanks for working on it. I will create a separate issue for redfish inventory #418

**Comment by bb-Ricardo** [comment]
As mentioned in the other issue. I currently lacking time and energy to work on this. therefore I will close this issue.
